qiskit/algorithms/time_evolvers/trotterization/trotter_qrte.py

```diff
@@ -0,0 +1,160 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021, 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""An algorithm to implement a Trotterization real time-evolution."""
+
+from __future__ import annotations
+
+from qiskit.algorithms.observables_evaluator import eval_observables
+
+from qiskit import QuantumCircuit
+from qiskit.algorithms.time_evolvers.evolution_problem import EvolutionProblem
+from qiskit.algorithms.time_evolvers.evolution_result import EvolutionResult
+from qiskit.algorithms.time_evolvers.real_evolver import RealEvolver
+from qiskit.opflow import (
+    CircuitOp,
+    PauliSumOp,
+    StateFn,
+)
+from qiskit.circuit.library import PauliEvolutionGate
+from qiskit.primitives import BaseEstimator
+from qiskit.quantum_info import Pauli
+from qiskit.synthesis import ProductFormula, LieTrotter
+
+
+class TrotterQRTE(RealEvolver):
+    """Quantum Real Time Evolution using Trotterization.
+    Type of Trotterization is defined by a ProductFormula provided.
+
+    Examples:
+
+        .. jupyter-execute::
+
+            from qiskit.opflow import X, Z, Zero
+            from qiskit.algorithms.time_evolvers import EvolutionProblem, TrotterQRTE
+
+            operator = X + Z
+            initial_state = Zero
+            time = 1
+            evolution_problem = EvolutionProblem(operator, 1, initial_state)
+            # LieTrotter with 1 rep
+            trotter_qrte = TrotterQRTE()
+            evolved_state = trotter_qrte.evolve(evolution_problem).evolved_state
+    """
+
+    def __init__(
+        self,
+        product_formula: ProductFormula | None = None,
+        estimator: BaseEstimator | None = None,
+    ) -> None:
+        """
+        Args:
+            product_formula: A Lie-Trotter-Suzuki product formula. The default is the Lie-Trotter
+                first order product formula with a single repetition.
+            estimator: An estimator primitive used for calculating expectation values of
+                EvolutionProblem.aux_operators.
+        """
+        if product_formula is None:
+            product_formula = LieTrotter()
+        self.product_formula = product_formula
```

Comment:
This is a public instance var, we should document it via Attributes so it exposed via the API ref. Any reason not to have estimator as a public instance var too and drop the setter/getter (dropping these and going for Attributes is something we have been trying to do recently and with the refactoring going on).

Reply:
As mentioned above we have a mix and match here. VQE, PVQD etc seem to have estimator as a public instance var. Since the setter getter here is trivial it would seem a public instance var would be more consistent with the other algos being redone and our desire to move away from having trivial getter/setters.